jupyter-extension/jupyterlab_pachyderm/tests/test_integrations.py

```diff
@@ -320,6 +321,58 @@ async def test_mount_datums(pachyderm_resources, http_client: AsyncClient):
     assert sorted([c["name"] for c in r.json()["content"]]) == sorted(files)
 
 
+async def test_mount_datums_multiple_batches(http_client: AsyncClient):
+    client = Client.from_config()
+    repo = pfs.Repo(name="multiple_batches")
+    client.pfs.delete_repo(repo=repo, force=True)
+    client.pfs.create_repo(repo=repo)
+
+    batch_size = DatumManager.DATUM_BATCH_SIZE
+    total_datums = int(batch_size * 1.5)
+    with client.pfs.commit(branch=pfs.Branch.from_uri(f"{repo.name}@master")) as c:
+        for i in range(total_datums):
+            # Number of files will require a request for a second batch of datums
+            c.put_file_from_bytes(path=f"/{i}", data=b"some data")
+    c.wait()
+
+    input_spec = {
+        "input": {
+            "pfs": {
+                "repo": repo.name,
+                "glob": "/*",
+            }
+        }
+    }
+    r = await http_client.put("/datums/_mount", json=input_spec)
+    assert r.status_code == 200, r.text
+    assert r.json()["idx"] == 0
+    assert r.json()["num_datums"] == batch_size
+    assert r.json()["all_datums_received"] is False
+
+    # Cycle to the last datum in the current batch
+    for i in range(1, batch_size):
+        r = await http_client.put("/datums/_next")
+        assert r.status_code == 200, r.text
+        assert r.json()["idx"] == i
+        assert r.json()["num_datums"] == batch_size
```

Comment:
I think `num_datums` has changed it's meaning. Before it meant "total number of datums" since `mount_datums` blocked until the `list_datum` completed. Now its "the total number of datums that have been streamed to the client". It might make sense to update the name

Reply:
That makes sense. What do you think of `num_datums_received`?